### DISCUSSION TITLE
IncompressibleFluidProperties interest?

### DISCUSSION THREAD

**Discussion by garcs2** [idaholab/moose#26856] [Q&A Modules: Thermal Hydraulics] [answered]
Hello,

Per my previous [discussion post](https://github.com/idaholab/moose/discussions/26791) I have now successfully implemented a routine to derive pressure from v and e (v -> rho -> T -> h then p = rho * (h-e)). However, I've now ran into an issue with implementing the e_from_v_h routine, since there's no way to backtrack pressure from any of those quantities. To this end, I was curious if there was any interest for me to create a IncompressibleFluidProperties Moose Object that only takes in temperature for most quantities, but pressure for quantities such as internal energy. From looking at the fluidproperties it doesn't seem that this would be redundant but of course would like other opinions.

**Comment by GiudGiud** [comment]
Hello

I dont know that a new base class for fluid properties will solve your problem. It seems like it would need to inherit from the current ones to be accessible in the current fluid flow solvers. 

I think what you want to do is implement the fluid solver that does not rely on quantities you cannot compute. Then all fluid properties can be queried with the existing calls even if 'pressure' is set to a dummy value for example

Guillaume

**Reply by garcs2** [reply]
Hi Guillaume,

Thank you for the response! When you say "implement a fluid solver that does not rely on quantities you cannot compute", where exactly would that correspond within the input file? Are you suggesting to change the fluid property type and use a substitute such as simplefluidproperties?

**Reply by GiudGiud** [reply]
Hello

No changing the fluid properties is an option and you can consider it. 
You can fit SimpleFP or StiffenedGasFP to be close to the fluid you are modeling, over a limited range of T, P. 

I was thinking of implementing a solver that does not rely on (v, e) but uses (p,T) in every equation

Guillaume

**Reply by garcs2** [reply]
Hi Guillaume,

I'm sorry I believe I need more clarification. So when you are referring to solver you are meaning going to the source file (NaKFluidProperties.C) and adjusting all the properties to take only pressure and temperature in every equation? 

I think I am confused since when I was originally trying to adopt the THM example to utilize NaK, all of the properties as functions of (p,T) were already implemented but would still get an error thrown at me per the [first discussion post](https://github.com/idaholab/moose/discussions/26791). So after I implemented that function, I would still get errors thrown for properties that were not implemented as functions of (v,e) that I then implemented myself (T, c, cp, cv, k, mu). My point is, NaKFluidProperties.C already has properties in terms of (p,T) so I'm not sure what I can do to circumvent getting errors thrown at me if that makes sense.

**Reply by GiudGiud** [reply, marked as answer]
Hello

No I mean re-writing the components in THM to use P,T instead of v,e
It s not a small task. But for incompressible fluids, I dont have another idea.

Guillaume
